tracing-jersey/src/test/java/com/palantir/tracing/jersey/TraceEnrichingFilterTest.java

```diff
@@ -156,6 +157,36 @@ public void testTraceState_withoutRequestHeadersGeneratesValidTraceResponseHeade
         assertThat(spanCaptor.getValue().getOperation(), is("GET /trace"));
     }
 
+    @Test
+    public void testTraceState_withoutRequestHeadersGeneratesValidTraceResponseHeadersWhenFailing() {
+        Response response = target.path("/failing-trace").request().get();
+        assertThat(response.getHeaderString(TraceHttpHeaders.TRACE_ID), not(nullValue()));
```

Comment:
taking a step back, it's not clear to me that the response needs to contain a trace id header at all (even in the success case).

Reply:
In the general case of streaming one? I'm confused, this is how it's been working so far everywhere. If not provided on the request headers, a trace will be created and returned so that you can find out what happens. This makes even more sense IMHO when things go wrong.

For what it's worth, here even forcing the trace with request headers is ignored in the failing streaming case. Doesn't really matter if you force it or let it pick, the exception handling breaks it.

Reply:
cool, that sounds reasonable.